### PR TITLE
conformance between spec and API: kmeans_init

### PR DESCRIPTION
**Pull request by mchernov-intel**
- default method: dense
- Task for `compute_input`, `compute_result` and `compute(...)`

### CODE REVIEW

File: cpp/oneapi/dal/algo/kmeans_init/compute_types.hpp [hunk 1]
```diff
@@ -21,12 +21,18 @@
...
 namespace detail {
+template <typename Task = task::by_default>
 class compute_input_impl;
+
+template <typename Task = task::by_default>
 class compute_result_impl;
 } // namespace detail
 
+template <typename Task = task::by_default>
 class ONEAPI_DAL_EXPORT compute_input : public base {
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/kmeans_init/compute_types.hpp:32]: Is it really depend on Task type in theory?

File: cpp/oneapi/dal/algo/kmeans_init/common.hpp [hunk 2]
```diff
@@ -21,8 +21,14 @@
 
 namespace oneapi::dal::kmeans_init {
 
+namespace task {
+struct init {};
```
[the remark this answers is not in the document]
Comment (mchernov-intel) [reply cpp/oneapi/dal/algo/kmeans_init/common.hpp:25]: resolved offline

File: cpp/oneapi/dal/algo/kmeans_init/common.hpp [hunk 3]
```diff
@@ -31,12 +37,13 @@ struct dense {};
 struct random_dense {};
 struct plus_plus_dense {};
 struct parallel_plus_dense {};
-using by_default = random_dense;
+using by_default = dense;
```
[the remark this answers is not in the document]
Comment (mchernov-intel) [reply cpp/oneapi/dal/algo/kmeans_init/common.hpp:40]: resolved offline

File: cpp/oneapi/dal/algo/kmeans_init/compute_types.hpp (hunk 1 above)
Comment (mchernov-intel) [reply cpp/oneapi/dal/algo/kmeans_init/compute_types.hpp:32]: should be discussed with @rlnx

Comment (mchernov-intel) [reply cpp/oneapi/dal/algo/kmeans_init/compute_types.hpp:32]: resolved

**Comment by mchernov-intel** [comment]
@michael-smirnov please, review

**Comment by michael-smirnov** [comment]
@mchernov-intel please fix internal ci (copyrights) and rebase to master to avoid problem with distributed kmeans.

Don't forget to merge it to the gold branch.

**Review by michael-smirnov — APPROVED** [review]
Approved with cond of green CI

**Pull request merged**